custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,287 @@
+            writer.writeheader()
+            for person_case_id in self._get_person_case_ids_to_process():
+                person_case = accessor.get_case(person_case_id)
+                if person_case.dynamic_case_properties().get('case_version') == PERSON_CASE_2B_VERSION:
+                    # reset all collections for next iteration
+                    occurrence_cases = []
+
+                    if "person" in case_types:
+                        case_status = self.get_case_status(person_case)
+                        case_status = self.update_case(person_case.get_id, case_status)
+                        writer.writerow(case_status)
+
+                    if "episode" in case_types:
+                        occurrence_cases = [case for case in accessor.get_reverse_indexed_cases([person_case_id])
+                                            if case.type == CASE_TYPE_OCCURRENCE]
+                        for occurrence_case in occurrence_cases:
+                            episode_cases = [case for case in
+                                             accessor.get_reverse_indexed_cases([occurrence_case.get_id])
+                                             if case.type == CASE_TYPE_EPISODE]
+                            for episode_case in episode_cases:
+                                case_status = self.get_case_status(episode_case, episode_cases=episode_cases)
+                                case_status = self.update_case(episode_case.get_id, case_status)
+                                writer.writerow(case_status)
+
+                    if "test" in case_types:
+                        if not occurrence_cases:
+                            occurrence_cases = [case for case in
+                                                accessor.get_reverse_indexed_cases([person_case_id])
+                                                if case.type == CASE_TYPE_OCCURRENCE]
+                        for occurrence_case in occurrence_cases:
+                            test_cases = [case for case in
+                                          accessor.get_reverse_indexed_cases([occurrence_case.get_id])
+                                          if case.type == CASE_TYPE_TEST]
+                            recently_opened_episode_case = None
+                            for test_case in test_cases:
+                                # For tests that are DSTB follow up tests, rft_general = follow_up_dstb
+                                # and result_recorded = yes
+                                recently_opened_episode_case, case_status = self.get_case_status(
+                                    test_case, recently_opened_episode_case=recently_opened_episode_case,
+                                    occurrence_case=occurrence_case)
+                                case_status = self.update_case(test_case.get_id, case_status)
+                                writer.writerow(case_status)
```

Comment:
How does `writer.writerow` handle nested dicts?  Does the `debug` information just show up as a json blob in a cell somewhere?

Reply:
Ya. It just shows as a dict. For ex:
```
{'debug':   {'treatment_initiation_date': u'2016-12-19', 'date_tested': u'2017-02-07',   'episode_case_id': u'82ed063b618b4f65bb193852f2caff63'},   'rft_dstb_follow_up_treatment_month': 2, 'datamigration_20_to_21': 'yes'}
--
```
Is a cell.